sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -154,6 +155,22 @@ class CacheManager extends Logging {
     }
   }
 
+  // Analyzes column statistics in the given cache data
+  private[sql] def analyzeColumnCacheQuery(
+      sparkSession: SparkSession,
+      cachedData: CachedData,
+      column: Seq[Attribute]): Unit = {
```

Comment:
@maropu . Recently, https://github.com/apache/spark/pull/24028 changed `cachedData` as an immutable sequence and this is guarded by `this` for all WRITE operations. This is a write operation, isn't it?

cc @cloud-fan